2019-02-14 10:27:23 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Syntax error in SQL statement "COUNT(*) from TABLES WHERE NAME = [*]'CREDENTIALS;"; SQL statement:
COUNT(*) from TABLES WHERE NAME = 'CREDENTIALS; [42000-197]
2019-02-14 10:28:01 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Syntax error in SQL statement "COUNT[*](*) FROM TABLES WHERE NAME = 'CREDENTIALS'; "; expected "COMMIT, CREATE, CALL, CHECKPOINT, COMMENT, {"; SQL statement:
COUNT(*) from TABLES WHERE NAME = 'CREDENTIALS'; [42001-197]
2019-02-14 10:28:48 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Table "TABLES" not found; SQL statement:
SELECT COUNT(*) FROM TABLES WHERE NAME = 'CREDENTIALS'; [42102-197]
2019-02-14 14:03:31 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Table "CREDENTIALS" already exists; SQL statement:
CREATE TABLE Credentials (	id INT NOT NULL AUTO_INCREMENT,	login VARCHAR(20) NOT NULL UNIQUE,	hash BINARY(256) NOT NULL,	salt BINARY(160) NOT NULL,	PRIMARY KEY(login)); [42101-197]
2019-02-14 14:08:50 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Table "CREDENTIALS" already exists; SQL statement:
CREATE TABLE Credentials (	id INT NOT NULL AUTO_INCREMENT,	login VARCHAR(20) NOT NULL UNIQUE,	hash BINARY(256) NOT NULL,	salt BINARY(160) NOT NULL,	PRIMARY KEY(login)); [42101-197]
2019-02-14 14:09:14 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Table "CREDENTIALS" already exists; SQL statement:
CREATE TABLE Credentials (	id INT NOT NULL AUTO_INCREMENT,	login VARCHAR(20) NOT NULL UNIQUE,	hash BINARY(256) NOT NULL,	salt BINARY(160) NOT NULL,	PRIMARY KEY(login)); [42101-197]
2019-02-14 14:14:12 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Table "CREDENTIALS" already exists; SQL statement:
CREATE TABLE Credentials (	id INT NOT NULL AUTO_INCREMENT,	login VARCHAR(20) NOT NULL UNIQUE,	hash BINARY(256) NOT NULL,	salt BINARY(160) NOT NULL,	PRIMARY KEY(login)); [42101-197]
2019-02-14 14:17:39 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Column "STREET" not found; SQL statement:
INSERT INTO Address (street, number, zip, city, country)		VALUES('?', ?, ?, '?', '?'); [42122-197]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:357)
	at org.h2.message.DbException.get(DbException.java:179)
	at org.h2.message.DbException.get(DbException.java:155)
	at org.h2.table.Table.getColumn(Table.java:682)
	at org.h2.command.Parser.parseColumn(Parser.java:936)
	at org.h2.command.Parser.parseColumnList(Parser.java:920)
	at org.h2.command.Parser.parseInsertGivenTable(Parser.java:1258)
	at org.h2.command.Parser.parseInsert(Parser.java:1222)
	at org.h2.command.Parser.parsePrepared(Parser.java:431)
	at org.h2.command.Parser.parse(Parser.java:335)
	at org.h2.command.Parser.parse(Parser.java:307)
	at org.h2.command.Parser.prepareCommand(Parser.java:278)
	at org.h2.engine.Session.prepareLocal(Session.java:611)
	at org.h2.engine.Session.prepareCommand(Session.java:549)
	at org.h2.jdbc.JdbcConnection.prepareCommand(JdbcConnection.java:1247)
	at org.h2.jdbc.JdbcPreparedStatement.<init>(JdbcPreparedStatement.java:76)
	at org.h2.jdbc.JdbcConnection.prepareStatement(JdbcConnection.java:304)
	at ch.b2btec.db.h2.AddressTable.insertAddress(AddressTable.java:61)
	at ch.b2btec.db.h2.Database.writeCustomer(Database.java:52)
	at ch.b2btec.bl.services.impl.UserManagementService.lambda$2(UserManagementService.java:33)
	at java.base/java.lang.Iterable.forEach(Iterable.java:75)
	at ch.b2btec.bl.services.impl.UserManagementService.lambda$1(UserManagementService.java:33)
	at java.base/java.util.Optional.ifPresent(Optional.java:183)
	at ch.b2btec.bl.services.impl.UserManagementService.load(UserManagementService.java:33)
	at ch.b2btec.ApplicationContext.<init>(ApplicationContext.java:38)
	at ch.b2btec.Shop.run(Shop.java:29)
	at ch.b2btec.Application$1.run(Application.java:25)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:313)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:770)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:715)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:740)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
2019-02-14 14:19:42 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Invalid value "3" for parameter "parameterIndex" [90008-197]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:357)
	at org.h2.message.DbException.get(DbException.java:179)
	at org.h2.message.DbException.getInvalidValueException(DbException.java:240)
	at org.h2.jdbc.JdbcPreparedStatement.setParameter(JdbcPreparedStatement.java:1560)
	at org.h2.jdbc.JdbcPreparedStatement.setInt(JdbcPreparedStatement.java:406)
	at ch.b2btec.db.h2.AddressTable.insertAddress(AddressTable.java:66)
	at ch.b2btec.db.h2.Database.writeCustomer(Database.java:52)
	at ch.b2btec.bl.services.impl.UserManagementService.lambda$2(UserManagementService.java:33)
	at java.base/java.lang.Iterable.forEach(Iterable.java:75)
	at ch.b2btec.bl.services.impl.UserManagementService.lambda$1(UserManagementService.java:33)
	at java.base/java.util.Optional.ifPresent(Optional.java:183)
	at ch.b2btec.bl.services.impl.UserManagementService.load(UserManagementService.java:33)
	at ch.b2btec.ApplicationContext.<init>(ApplicationContext.java:38)
	at ch.b2btec.Shop.run(Shop.java:29)
	at ch.b2btec.Application$1.run(Application.java:25)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:313)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:770)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:715)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:740)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
2019-02-14 14:21:29 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Method is only allowed for a query. Use execute or executeUpdate instead of executeQuery; SQL statement:
INSERT INTO Address (street, number, zip, city, country)		VALUES(?, ?, ?, ?, ?); [90002-197]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:357)
	at org.h2.message.DbException.get(DbException.java:179)
	at org.h2.message.DbException.get(DbException.java:155)
	at org.h2.message.DbException.get(DbException.java:144)
	at org.h2.command.Prepared.query(Prepared.java:219)
	at org.h2.command.CommandContainer.query(CommandContainer.java:114)
	at org.h2.command.Command.executeQuery(Command.java:202)
	at org.h2.jdbc.JdbcPreparedStatement.executeQuery(JdbcPreparedStatement.java:114)
	at ch.b2btec.db.h2.AddressTable.insertAddress(AddressTable.java:69)
	at ch.b2btec.db.h2.Database.writeCustomer(Database.java:52)
	at ch.b2btec.bl.services.impl.UserManagementService.lambda$2(UserManagementService.java:33)
	at java.base/java.lang.Iterable.forEach(Iterable.java:75)
	at ch.b2btec.bl.services.impl.UserManagementService.lambda$1(UserManagementService.java:33)
	at java.base/java.util.Optional.ifPresent(Optional.java:183)
	at ch.b2btec.bl.services.impl.UserManagementService.load(UserManagementService.java:33)
	at ch.b2btec.ApplicationContext.<init>(ApplicationContext.java:38)
	at ch.b2btec.Shop.run(Shop.java:29)
	at ch.b2btec.Application$1.run(Application.java:25)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:313)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:770)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:715)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:740)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
2019-02-14 14:38:35 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Table "CREDENTIALSWHERE" not found; SQL statement:
SELECT	COUNT(*)FROM	CredentialsWHERE	login = ?; [42102-197]
2019-02-15 09:25:24 database: flush
org.h2.message.DbException: General error: "java.lang.IllegalStateException: The file is locked: nio:C:/Users/Thomas/mas-workspace/B2B-tec week 5 solution/database/b2b-tec.mv.db [1.4.197/7]" [50000-197]
	at org.h2.message.DbException.get(DbException.java:168)
	at org.h2.message.DbException.convert(DbException.java:307)
	at org.h2.mvstore.db.MVTableEngine$1.uncaughtException(MVTableEngine.java:95)
	at org.h2.mvstore.MVStore.handleException(MVStore.java:2506)
	at org.h2.mvstore.MVStore.panic(MVStore.java:383)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:364)
	at org.h2.mvstore.MVStore$Builder.open(MVStore.java:2934)
	at org.h2.mvstore.db.MVTableEngine$Store.open(MVTableEngine.java:155)
	at org.h2.mvstore.db.MVTableEngine.init(MVTableEngine.java:100)
	at org.h2.engine.Database.getPageStore(Database.java:2538)
	at org.h2.engine.Database.open(Database.java:709)
	at org.h2.engine.Database.openDatabase(Database.java:286)
	at org.h2.engine.Database.<init>(Database.java:280)
	at org.h2.engine.Engine.openSession(Engine.java:66)
	at org.h2.engine.Engine.openSession(Engine.java:179)
	at org.h2.engine.Engine.createSessionAndValidate(Engine.java:157)
	at org.h2.engine.Engine.createSession(Engine.java:140)
	at org.h2.engine.Engine.createSession(Engine.java:28)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:351)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:124)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:103)
	at org.h2.Driver.connect(Driver.java:69)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:677)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:228)
	at ch.b2btec.db.h2.Database.<init>(Database.java:34)
	at ch.b2btec.db.h2.Database.getInstance(Database.java:19)
	at ch.b2btec.bl.services.impl.UserManagementService.load(UserManagementService.java:32)
	at ch.b2btec.ApplicationContext.<init>(ApplicationContext.java:38)
	at ch.b2btec.Shop.run(Shop.java:29)
	at ch.b2btec.Application$1.run(Application.java:25)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:313)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:770)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:715)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:740)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
Caused by: org.h2.jdbc.JdbcSQLException: General error: "java.lang.IllegalStateException: The file is locked: nio:C:/Users/Thomas/mas-workspace/B2B-tec week 5 solution/database/b2b-tec.mv.db [1.4.197/7]" [50000-197]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:357)
	... 43 more
Caused by: java.lang.IllegalStateException: The file is locked: nio:C:/Users/Thomas/mas-workspace/B2B-tec week 5 solution/database/b2b-tec.mv.db [1.4.197/7]
	at org.h2.mvstore.DataUtils.newIllegalStateException(DataUtils.java:870)
	at org.h2.mvstore.FileStore.open(FileStore.java:173)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:350)
	... 37 more
2019-02-15 09:26:10 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Table "CREDENTIALSSET" not found; SQL statement:
UPDATE	CredentialsSET	hash = ?,	salt = ? WHERE	login = ?; [42102-197]
2019-02-15 10:53:24 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Syntax error in SQL statement "SELECT * FROM ADDRESS CREDENTIALS CREDENTIALS[*]"; SQL statement:
SELECT * FROM ADDRESS CREDENTIALS CREDENTIALS [42000-197]
2019-02-15 10:54:07 database: flush
org.h2.message.DbException: General error: "java.lang.IllegalStateException: The file is locked: nio:C:/Users/Thomas/mas-workspace/B2B-tec week 5 solution/database/b2b-tec.mv.db [1.4.197/7]" [50000-197]
	at org.h2.message.DbException.get(DbException.java:168)
	at org.h2.message.DbException.convert(DbException.java:307)
	at org.h2.mvstore.db.MVTableEngine$1.uncaughtException(MVTableEngine.java:95)
	at org.h2.mvstore.MVStore.handleException(MVStore.java:2506)
	at org.h2.mvstore.MVStore.panic(MVStore.java:383)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:364)
	at org.h2.mvstore.MVStore$Builder.open(MVStore.java:2934)
	at org.h2.mvstore.db.MVTableEngine$Store.open(MVTableEngine.java:155)
	at org.h2.mvstore.db.MVTableEngine.init(MVTableEngine.java:100)
	at org.h2.engine.Database.getPageStore(Database.java:2538)
	at org.h2.engine.Database.open(Database.java:709)
	at org.h2.engine.Database.openDatabase(Database.java:286)
	at org.h2.engine.Database.<init>(Database.java:280)
	at org.h2.engine.Engine.openSession(Engine.java:66)
	at org.h2.engine.Engine.openSession(Engine.java:179)
	at org.h2.engine.Engine.createSessionAndValidate(Engine.java:157)
	at org.h2.engine.Engine.createSession(Engine.java:140)
	at org.h2.engine.Engine.createSession(Engine.java:28)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:351)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:124)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:103)
	at org.h2.Driver.connect(Driver.java:69)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:677)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:228)
	at ch.b2btec.db.h2.Database.<init>(Database.java:35)
	at ch.b2btec.db.h2.Database.getInstance(Database.java:20)
	at ch.b2btec.bl.services.impl.UserManagementService.load(UserManagementService.java:36)
	at ch.b2btec.ApplicationContext.<init>(ApplicationContext.java:38)
	at ch.b2btec.Shop.run(Shop.java:29)
	at ch.b2btec.Application$1.run(Application.java:25)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:313)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:770)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:715)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:740)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
Caused by: org.h2.jdbc.JdbcSQLException: General error: "java.lang.IllegalStateException: The file is locked: nio:C:/Users/Thomas/mas-workspace/B2B-tec week 5 solution/database/b2b-tec.mv.db [1.4.197/7]" [50000-197]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:357)
	... 43 more
Caused by: java.lang.IllegalStateException: The file is locked: nio:C:/Users/Thomas/mas-workspace/B2B-tec week 5 solution/database/b2b-tec.mv.db [1.4.197/7]
	at org.h2.mvstore.DataUtils.newIllegalStateException(DataUtils.java:870)
	at org.h2.mvstore.FileStore.open(FileStore.java:173)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:350)
	... 37 more
2019-02-15 10:54:38 database: flush
org.h2.message.DbException: General error: "java.lang.IllegalStateException: The file is locked: nio:C:/Users/Thomas/mas-workspace/B2B-tec week 5 solution/database/b2b-tec.mv.db [1.4.197/7]" [50000-197]
	at org.h2.message.DbException.get(DbException.java:168)
	at org.h2.message.DbException.convert(DbException.java:307)
	at org.h2.mvstore.db.MVTableEngine$1.uncaughtException(MVTableEngine.java:95)
	at org.h2.mvstore.MVStore.handleException(MVStore.java:2506)
	at org.h2.mvstore.MVStore.panic(MVStore.java:383)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:364)
	at org.h2.mvstore.MVStore$Builder.open(MVStore.java:2934)
	at org.h2.mvstore.db.MVTableEngine$Store.open(MVTableEngine.java:155)
	at org.h2.mvstore.db.MVTableEngine.init(MVTableEngine.java:100)
	at org.h2.engine.Database.getPageStore(Database.java:2538)
	at org.h2.engine.Database.open(Database.java:709)
	at org.h2.engine.Database.openDatabase(Database.java:286)
	at org.h2.engine.Database.<init>(Database.java:280)
	at org.h2.engine.Engine.openSession(Engine.java:66)
	at org.h2.engine.Engine.openSession(Engine.java:179)
	at org.h2.engine.Engine.createSessionAndValidate(Engine.java:157)
	at org.h2.engine.Engine.createSession(Engine.java:140)
	at org.h2.engine.Engine.createSession(Engine.java:28)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:351)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:124)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:103)
	at org.h2.Driver.connect(Driver.java:69)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:677)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:228)
	at ch.b2btec.db.h2.Database.<init>(Database.java:35)
	at ch.b2btec.db.h2.Database.getInstance(Database.java:20)
	at ch.b2btec.bl.services.impl.UserManagementService.load(UserManagementService.java:36)
	at ch.b2btec.ApplicationContext.<init>(ApplicationContext.java:38)
	at ch.b2btec.Shop.run(Shop.java:29)
	at ch.b2btec.Application$1.run(Application.java:25)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:313)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:770)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:715)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:740)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
Caused by: org.h2.jdbc.JdbcSQLException: General error: "java.lang.IllegalStateException: The file is locked: nio:C:/Users/Thomas/mas-workspace/B2B-tec week 5 solution/database/b2b-tec.mv.db [1.4.197/7]" [50000-197]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:357)
	... 43 more
Caused by: java.lang.IllegalStateException: The file is locked: nio:C:/Users/Thomas/mas-workspace/B2B-tec week 5 solution/database/b2b-tec.mv.db [1.4.197/7]
	at org.h2.mvstore.DataUtils.newIllegalStateException(DataUtils.java:870)
	at org.h2.mvstore.FileStore.open(FileStore.java:173)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:350)
	... 37 more
2019-02-15 11:01:28 database: flush
org.h2.message.DbException: General error: "java.lang.IllegalStateException: The file is locked: nio:C:/Users/Thomas/mas-workspace/B2B-tec week 5 solution/database/b2b-tec.mv.db [1.4.197/7]" [50000-197]
	at org.h2.message.DbException.get(DbException.java:168)
	at org.h2.message.DbException.convert(DbException.java:307)
	at org.h2.mvstore.db.MVTableEngine$1.uncaughtException(MVTableEngine.java:95)
	at org.h2.mvstore.MVStore.handleException(MVStore.java:2506)
	at org.h2.mvstore.MVStore.panic(MVStore.java:383)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:364)
	at org.h2.mvstore.MVStore$Builder.open(MVStore.java:2934)
	at org.h2.mvstore.db.MVTableEngine$Store.open(MVTableEngine.java:155)
	at org.h2.mvstore.db.MVTableEngine.init(MVTableEngine.java:100)
	at org.h2.engine.Database.getPageStore(Database.java:2538)
	at org.h2.engine.Database.open(Database.java:709)
	at org.h2.engine.Database.openDatabase(Database.java:286)
	at org.h2.engine.Database.<init>(Database.java:280)
	at org.h2.engine.Engine.openSession(Engine.java:66)
	at org.h2.engine.Engine.openSession(Engine.java:179)
	at org.h2.engine.Engine.createSessionAndValidate(Engine.java:157)
	at org.h2.engine.Engine.createSession(Engine.java:140)
	at org.h2.engine.Engine.createSession(Engine.java:28)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:351)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:124)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:103)
	at org.h2.Driver.connect(Driver.java:69)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:677)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:228)
	at ch.b2btec.db.h2.Database.<init>(Database.java:35)
	at ch.b2btec.db.h2.Database.getInstance(Database.java:20)
	at ch.b2btec.bl.services.impl.UserManagementService.load(UserManagementService.java:36)
	at ch.b2btec.ApplicationContext.<init>(ApplicationContext.java:38)
	at ch.b2btec.Shop.run(Shop.java:29)
	at ch.b2btec.Application$1.run(Application.java:25)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:313)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:770)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:715)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:740)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
Caused by: org.h2.jdbc.JdbcSQLException: General error: "java.lang.IllegalStateException: The file is locked: nio:C:/Users/Thomas/mas-workspace/B2B-tec week 5 solution/database/b2b-tec.mv.db [1.4.197/7]" [50000-197]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:357)
	... 43 more
Caused by: java.lang.IllegalStateException: The file is locked: nio:C:/Users/Thomas/mas-workspace/B2B-tec week 5 solution/database/b2b-tec.mv.db [1.4.197/7]
	at org.h2.mvstore.DataUtils.newIllegalStateException(DataUtils.java:870)
	at org.h2.mvstore.FileStore.open(FileStore.java:173)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:350)
	... 37 more
